### PR TITLE
LibRaw update

### PR DESCRIPTION
**Pull request by kmilos**
@TurboGit This is just a heads up, 0.21.0 final should have been released by now (using head/RC2 for now), would be good to include the stable lib for dt 4.2...

Also turned off some unused LibRaw features for a faster/smaller build.

### CODE REVIEW

**Comment by kmilos** [comment]
[LibRaw 0.21.0 was released recently.](https://www.libraw.org/news/libraw-0-21-release)

**Comment by kmilos** [comment]
Adding an option to unbundle the external/LibRaw as [0.21.0 is now available in quite a few places.](https://repology.org/project/libraw/versions)

**Comment by TurboGit** [comment]
Probably worth having that in 4.2.1, should be safe, right?

**Comment by kmilos** [comment]
Yeah, better than having some RC code in there, there's been a lot of OSS-Fuzz related fixes. But please review the "unbundle" option more closely in that case...

**Comment by TurboGit** [comment]
> But please review the "unbundle" option more closely in that case...

This is when linking to an OS installed packaged outside of darktable, right?

**Comment by kmilos** [comment]
> This is when linking to an OS installed packaged outside of darktable, right?

Yes. We had it bundled because official 0.21 that has support for all new Canon cameras was not released and not available in any distro, which is now slowly [changing](https://repology.org/project/libraw/versions).

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -412,16 +412,19 @@ if(USE_HEIF)
...
-#    find_package(libraw 0.20.2)
-#    if (libraw_FOUND)
-#        list(APPEND LIBS ${libraw_LIBRARY})
-#        add_definitions(-DHAVE_LIBRAW=1)
-#        list(APPEND SOURCES "common/imageio_libraw.c")
-#    endif()
-   add_definitions(-DHAVE_LIBRAW=1)
-   list(APPEND SOURCES "common/imageio_libraw.c")
+# For now we use the LibRaw submodule
+if(USE_LIBRAW)
```
Comment (TurboGit) [inline src/CMakeLists.txt:416]: Can we add some status message here:
1. saying the bundled libraw is being used
2. or saying that the OS libraw is being used

Comment (kmilos) [reply src/CMakeLists.txt:416]: Good idea, will do.

Btw, what should be the default? Prefer the system lib if available, or intree copy (as is currently)?

Comment (TurboGit) [reply src/CMakeLists.txt:416]: I would stay on intree copy for now until we have more visibility on which distribs have updated LibRaw. And we could reconsider the default later.

**Review by TurboGit — APPROVED** [review]
Working for me. Well I have OS version 0.20, after tweaking to allow this version I got an expected compilation error as this version is too old... But at least this means that using `-DDONT_USE_INTERNAL_LIBRAW=ON` works as expected.

**Comment by TurboGit** [comment]
Merged in master & 4.2 branch.

@kmilos : Can you put there a little blurb that I'll add in the release note for what is new... If there is anything important to say. TIA.

**Comment by kmilos** [comment]
Something like:

"For systems providing LibRaw 0.21.0 or newer, it is now possible to disable building the bundled copy by defining `-DDONT_USE_INTERNAL_LIBRAW=ON`"

**Pull request merged**